### PR TITLE
metrological_agents class updated

### PR DESCRIPTION
**Pull request by anupam-prasad**
#131  resolved here. The MetrologicalMonitorAgent still inherits from MetrologicalAgent and not MonitorAgent. The class was instead rewritten to to be functionally similar to MonitorAgent. 

The on_received_message method now explicitly adds the metadata to the buffer.

### CODE REVIEW

**Comment by bangxiangyong** [comment]
Fantastic job, i'll try it out asap

File: agentMET4FOF/agents.py [hunk 1]
```diff
@@ -1611,7 +1614,8 @@ class MonitorAgent(AgentMET4FOF):
         Used to specifically select only a few keys to be plotted
     """
 
-    def init_parameters(self, plot_filter=[], custom_plot_function=-1, **kwargs):
+    def init_parameters(self, plot_filter=[], custom_plot_function=-1, *args, **kwargs):
+        # super(MonitorAgent, self).init_parameters()
```
Comment (bangxiangyong) [inline agentMET4FOF/agents.py:1618]: this is a good intention. however, for the *args to work upon instantiation of agents, we will need to add it to the _add_osbrain_agent and _add_mesa_agent functions and pass the *args appropriately. Inside these functions, init_parameters are also called.

Comment (bangxiangyong) [reply agentMET4FOF/agents.py:1618]: if you could remove the commented super method, that'd be great as well to reduce clutter

File: agentMET4FOF/metrological_agents.py [hunk 2]
```diff
@@ -146,5 +193,17 @@ def custom_plot_function(self, data, sender_agent, **kwargs):
...
+        # ut=.001*np.ones_like(t)
+        # uv=.005*np.ones_like(v)
+        # trace = go.Scatter(
+        #     x=t,
+        #     y=v,
+        #     error_x=dict(type="data", array=ut, visible=True),
+        #     error_y=dict(type="data", array=uv, visible=True),
+        #     mode="lines",
+        #     name=f"blah blah",
+        # )
```
Comment (bangxiangyong) [inline agentMET4FOF/metrological_agents.py:208]: again if the unused comments can be removed from the final code, thatd be great. thanks

File: agentMET4FOF_tutorials/tutorial_1_generator_agent.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def demonstrate_generator_agent_use():
 
     # Interconnect agents by either way:
     # 1) by agent network.bind_agents(source, target).
-    agent_network.bind_agents(gen_agent, monitor_agent)
+   # agent_network.bind_agents(gen_agent, monitor_agent)
```
Comment (bangxiangyong) [inline agentMET4FOF_tutorials/tutorial_1_generator_agent.py:44]: the tutorial 1 should be left as it is

**Review by bangxiangyong — CHANGES_REQUESTED** [review]
1. great job in fixing tutorial 4 and the metrological agents..! i've ran the code and appears to be working now 👍 
2. however, there appears to be some unusued comments which can be cleaned further before merging and i've commented them below.

File: agentMET4FOF/metrological_agents.py (hunk 2 above)
Comment (anupam-prasad) [reply agentMET4FOF/metrological_agents.py]: Done. There were a few commented lines in the original code in the MetrologicalAgents class. @mgrub Can these be removed as well? Lines 11-15 and 19-23.

File: agentMET4FOF/agents.py (hunk 1 above)
Comment (anupam-prasad) [reply agentMET4FOF/agents.py]: I've removed the commented code. Should the *args be removed for now as well?

**Review by anupam-prasad — COMMENTED** [review]
Removed commented code and unused imports.

File: agentMET4FOF/metrological_agents.py [hunk 4]
```diff
@@ -106,34 +107,80 @@ def pack_data(self, data, channel="default"):
 
 class MetrologicalMonitorAgent(MetrologicalAgent):
     def init_parameters(self, *args, **kwargs):
-        super().init_parameters(*args, **kwargs)
-
+        super(MetrologicalMonitorAgent, self).init_parameters(*args, **kwargs)
+        #super().init_parameters(*args, **kwargs)
```
Comment (bangxiangyong) [inline agentMET4FOF/metrological_agents.py:111]: just one more unused comment here and it should be good to go !

**Review by bangxiangyong — CHANGES_REQUESTED** [review]

File: agentMET4FOF/metrological_agents.py (hunk 4 above)
Comment (anupam-prasad) [reply agentMET4FOF/metrological_agents.py]: Done!

**Pull request merged**